server/plugin_test.go

```diff
@@ -64,23 +63,76 @@ func setupTestPlugin(t *testing.T, api *plugintest.API, siteURL string) *Matterp
 		},
 	}
 	p.SetAPI(api)
-	err := p.OnActivate()
-	require.Nil(t, err)
+	p.router = p.InitAPI()
+
 	return p
 }
 
 func TestPluginOnActivate(t *testing.T) {
-	p := &MatterpollPlugin{
-		Config: &Config{},
+	t.Run("empty config", func(t *testing.T) {
+		api := &plugintest.API{}
+		api.On("GetServerVersion").Return("5.4.0")
+		p := &MatterpollPlugin{}
+		p.SetAPI(api)
+
+		err := p.OnActivate()
+
+		assert.NotNil(t, err)
+	})
+
+	for name, test := range map[string]struct {
+		SetupAPI    func(*plugintest.API) *plugintest.API
+		ShouldError bool
+	}{
+		"server version: 5.5.0": {
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
```

Comment:
`SetupAPI` in all cases looks same, so we can replace `SetupAPI func(*plugintest.API) *plugintest.API` field to `ServerVersion string` field.

Reply:
Sure we could do this. But I'm quite sure we will add more API calls to `OnActivate()`, so I would like to leave it as it is.

Reply:
Got it 👍